demo/api/src/menus/menus.resolver.ts

```diff
@@ -46,17 +47,8 @@ export class MenusResolver {
     }
 
     @Query(() => [PageTreeNode])
-    async topMenu(
-        @Args("scope", { type: () => PageTreeNodeScope }) scope: PageTreeNodeScope,
-        @RequestContext() { includeInvisiblePages }: RequestContextInterface,
-    ): Promise<PageTreeNodeInterface[]> {
-        return (
-            this.pageTreeService
-                .createReadApi({
-                    visibility: [PageTreeNodeVisibility.Published, ...(includeInvisiblePages || [])],
-                })
-                //.preloadNodes(scope)
-                .pageTreeRootNodeList({ scope, category: PageTreeNodeCategory.TopMenu, excludeHiddenInMenu: true })
-        );
+    async topMenu(@Args("scope", { type: () => PageTreeNodeScope }) scope: PageTreeNodeScope): Promise<PageTreeNodeInterface[]> {
+        this.pageTreeReadApi.preloadNodes(scope);
```

Comment:
I believe this may be an example where preloading does not improve performance: The top menu may consist of just a few nodes.

Another question: Should `preloadNodes` support `category` as well?

Reply:
yes, quite possible. I'll remove it and add a comment why